src/aggregation/bucket/histogram/date_histogram.rs

```diff
@@ -55,42 +66,144 @@ pub struct DateHistogramAggregationReq {
+
     /// Whether to return the buckets as a hash map
     #[serde(default)]
     pub keyed: bool,
 }
 
 impl DateHistogramAggregationReq {
+    pub(crate) fn to_histogram_req(&self) -> crate::Result<HistogramAggregation> {
+        self.validate()?;
+        Ok(HistogramAggregation {
+            field: self.field.to_string(),
+            interval: parse_into_microseconds(&self.fixed_interval)? as f64,
+            offset: self
+                .offset
+                .as_ref()
+                .map(|offset| parse_offset_into_microseconds(offset))
+                .transpose()?
+                .map(|el| el as f64),
+            min_doc_count: self.min_doc_count,
+            hard_bounds: None,
+            extended_bounds: None,
+            keyed: self.keyed,
+        })
+    }
+
     fn validate(&self) -> crate::Result<()> {
+        if self.interval.is_some() {
+            return Err(crate::TantivyError::InvalidArgument(format!(
+                "`interval` parameter {:?} in date histogram is unsupported, only \
+                 `fixed_interval` is supported",
+                self.interval
+            )));
+        }
+        if self.format.is_some() {
+            return Err(crate::TantivyError::InvalidArgument(
+                "format parameter on date_histogram is unsupported".to_string(),
+            ));
+        }
+
+        if self.date_interval.is_some() {
+            return Err(crate::TantivyError::InvalidArgument(
+                "date_interval in date histogram is unsupported, only `fixed_interval` is \
+                 supported"
+                    .to_string(),
+            ));
+        }
+
+        parse_into_microseconds(&self.fixed_interval)?;
+
         Ok(())
     }
 }
 
-#[derive(Debug, PartialEq, Eq)]
+#[derive(Debug, Clone, PartialEq, Eq, Error)]
 /// Errors when parsing the fixed interval for `DateHistogramAggregationReq`.
 pub enum DateHistogramParseError {
     /// Unit not recognized in passed String
+    #[error("Unit not recognized in passed String {0:?}")]
     UnitNotRecognized(String),
     /// Number not found in passed String
+    #[error("Number not found in passed String {0:?}")]
     NumberMissing(String),
     /// Unit not found in passed String
+    #[error("Unit not found in passed String {0:?}")]
     UnitMissing(String),
+    /// Offset invalid
+    #[error("passed offset is invalid {0:?}")]
+    InvalidOffset(String),
 }
 
-fn parse_into_milliseconds(input: &str) -> Result<u64, DateHistogramParseError> {
+fn parse_offset_into_microseconds(input: &str) -> Result<i64, AggregationError> {
+    let is_sign = |byte| &[byte] == b"-" || &[byte] == b"+";
+    if input.is_empty() {
+        return Err(DateHistogramParseError::InvalidOffset(input.to_string()).into());
+    }
+
+    let has_sign = is_sign(input.as_bytes()[0]);
+    if has_sign {
+        let (sign, input) = input.split_at(1);
+        let mut val = parse_into_microseconds(input)?;
+        if sign == "-" {
+            val *= -1;
```

Comment:
nitpick,

```
if sign == "-" {
   -val
} else {
   val
}
```

would have been a tiny bit more readable... But overall I am missing the point of the whole fucntion.